synapse/handlers/auth.py

```diff
@@ -108,7 +108,11 @@ def __init__(self, hs):
 
         # Ratelimiter for failed auth during UIA. Uses same ratelimit config
         # as per `rc_login.failed_attempts`.
-        self._failed_uia_attempts_ratelimiter = Ratelimiter()
+        # XXX: Should this be hs.get_login_failed_attempts_ratelimiter?
```

Comment:
Probably not, UI auth is used for things that aren't login? I'm not sure it would matter though.

Reply:
They use the same configured `rate_hz` and `burst_count`. It would affect things from a user perspective though:

If they're two separate objects, you can hammer `/login` and then hammer some UIA endpoint.

If they're the same object, you can hammer `/login` but are then ratelimited as soon as you touch a UIA endpoint.

I think the former is probably more intuitive, but we may not want to give people a pass if they've ratelimited any endpoints.

Actually, that's a good point. If this PR splits up ratelimiters into different objects, then you won't be punished for hitting one ratelimit versus another. Thoughts?

Reply:
Ok, nevermind, we have the same number of ratelimiters. I thought a lot of classes were using `hs.get_ratelimiter()` but it was just one synapse class and a lot of tests.